README.md

```diff
@@ -0,0 +1,494 @@
+    * [logger.flush](#loggerflush)
+    * [logger.log](#loggerlogstatement-options)
+    * [logger.removeMetaProperty](#loggerremovemetapropertykey)
+* **[How Log Lines are Sent to LogDNA](#how-log-lines-are-sent-to-logdna)**
+* **[Exponential Backoff Strategy](#exponential-backoff-strategy)**
+* **[Best Practices](#best-practices)**
+* **[Client Side Support](#client-side)**
+* **[Bunyan Stream](#bunyan-stream)**
+* **[Winston Transport](#winston-transport)**
+* **[AWS Lambda Support](#aws-lambda-support)**
+* **[License](#license)**
+
+## Migrating From Other Versions
+
+Previous versions of this client are [still supported](https://github.com/logdna/nodejs), but if you are upgrading to this version, please see
+our [migration document](./docs/migrating-from-other-versions.md) for the differences between this and prior versions.
+
+## Install
+
+```javascript
+$ npm install @logdna/logger
+```
+
+## Setup
+Operation requires a [LogDNA Ingestion Key](https://docs.logdna.com/docs/ingestion-key).  Without it, the client will not be able to post
+logs to the cloud.  Please contact our support if you have questions about this!
+
+
+## Usage
+
+To use the client, simply create an instance, then call `.log()` or a [convenience method](#convenience-methods).
+
+```javascript
+const logdna = require('@logdna/logger')
+
+const options = {
+  app: 'myAppName'
+, level: 'debug' // set a default for when level is not provided in function calls
+}
+
+const logger = logdna.createLogger('<YOUR INGESTION KEY>', options)
+
+logger.log('This is an INFO statement', 'info')
+
+logger.log('This will be a DEBUG statement, based on the default')
+
+logger.log('This is an INFO statement with an options object', {
+  level: 'info'
+})
+
+logger.info('This is an INFO statement using a convenience method')
+
+logger.error('This is an error with meta data attached', {
+  indexMeta: true
+, meta: {
+    message: 'TypeError for XYZ'
+  , err
+  }
+})
+
+logger.error(err) // Objects can be logged too
+```
+
+## Supported Log Levels
+The client supports the following log levels.  They are case-insensitive.
+
+* `TRACE`
+* `DEBUG`
+* `INFO`
+* `WARN`
+* `ERROR`
+* `FATAL`
+
+## Convenience Methods
+We have set up convenience methods that automatically set the log level appropriately, and are easy to read.
+
+### `logger.trace(msg[, options])`
+### `logger.debug(msg[, options])`
+### `logger.info(msg[, options])`
+### `logger.warn(msg[, options])`
+### `logger.error(msg[, options])`
+### `logger.fatal(msg[, options])`
+
+* `msg` [`<Object>`][] | [`<String>`][] - The message (or object) to log
+* `options ` [`<Object>`][] - Per-message options.  See [`logger.log()`](#loggerlogstatement-options) for those.
+
+## Events
+The `Logger` is an `EventEmitter` and will emit events rather than use promises or callbacks to communicate its progress.  
+Listening to the events is optional, although an `error` listener is recommended.
+
+### Event: `'addMetaProperty'`
```

Comment:
should this be in past tense, @darinspivey? This event looks odd compared to others maybe because of its name. You can combine `addMetaProperty` and `removeMetaProperty` events under `meta` event having different `message` for each - I think, that'd be better.

Reply:
There's a few reasons why it's done this way.

1.  It helps the name of the event to match the function call rather than force people to change tenses
2. Combining events can be confusing because of #1, and because the data is different.  `addMetaProperty` has a `key` and a `value`, and that's different from `removeMetaProperty`.  I don't think combining does anything besides force the consumer to parse out which is which.

I'd prefer we leave it as-is.